lua/nvim_lsp/configs.lua

```diff
@@ -146,8 +147,15 @@ function configs.__newindex(t, config_name, config_def)
               ))
         end
       end)
+
+      new_config.root_dir = _root_dir
```

Comment:
Why did you move this process here? I don't think I need to move.

Reply:
because I use to call `validate_client_config` and it would complain about a missing root_dir. make_config should generate a full config and not a half decent one. Because I don't use `validate_client_config` anymore, I can now pass a fake path (lsp will validate it only upon starting lsp) so it solves the issue of having a valid path. Thanks for making me notice.